new-log-viewer/src/components/MenuBar.tsx

```diff
@@ -0,0 +1,158 @@
+import React, {
+    useContext,
+    useState,
+} from "react";
+
+import {
+    IconButton,
+    Sheet,
+    Typography,
+} from "@mui/joy";
+
+import {SvgIconComponent} from "@mui/icons-material";
+import Description from "@mui/icons-material/Description";
+import FileOpenIcon from "@mui/icons-material/FileOpen";
+import NavigateBefore from "@mui/icons-material/NavigateBefore";
+import NavigateNext from "@mui/icons-material/NavigateNext";
+import Settings from "@mui/icons-material/Settings";
+import SkipNext from "@mui/icons-material/SkipNext";
+import SkipPrevious from "@mui/icons-material/SkipPrevious";
+import TipsAndUpdates from "@mui/icons-material/TipsAndUpdates";
+
+import {StateContext} from "../contexts/StateContextProvider";
+import {
+    updateWindowUrlHashParams,
+    UrlContext,
+} from "../contexts/UrlContextProvider";
+import {CONFIG_KEY} from "../typings/config";
+import {CURSOR_CODE} from "../typings/worker";
+import {ACTION_NAME} from "../utils/actions";
+import {getConfig} from "../utils/config";
+import {openFile} from "../utils/file";
+import {
+    getFirstItemNumInNextChunk,
+    getLastItemNumInPrevChunk,
+} from "../utils/math";
+import ConfigModal from "./modals/SettingsModal";
+
+
+/**
+ *
+ * @param actionName
+ * @param logEventNum
+ * @param numEvents
+ */
+export const handleAction = (actionName: ACTION_NAME, logEventNum: number, numEvents: number) => {
+    const pageSize = getConfig(CONFIG_KEY.PAGE_SIZE);
+    switch (actionName) {
+        case ACTION_NAME.FIRST_PAGE:
+            updateWindowUrlHashParams({logEventNum: 1});
+            break;
+        case ACTION_NAME.PREV_PAGE:
+            updateWindowUrlHashParams({
+                logEventNum: getLastItemNumInPrevChunk(logEventNum, pageSize),
+            });
+            break;
+        case ACTION_NAME.NEXT_PAGE:
+            updateWindowUrlHashParams({
+                logEventNum: getFirstItemNumInNextChunk(logEventNum, pageSize),
+            });
+            break;
+        case ACTION_NAME.LAST_PAGE:
+            updateWindowUrlHashParams({logEventNum: numEvents});
+            break;
+        default:
+            break;
+    }
+};
+
+
+/**
+ *
+ */
+export const MenuBar = () => {
```

Comment:
If we are not exporting other variables in this source file, we can write `export default MenuBar` at the end of this source file.